pkg/api/db_controller_test.go

```diff
@@ -0,0 +1,2218 @@
+package api_test
```

Comment:
Copying this amount of code, even if it's a test code, would lead to a huge maintenance overhead and possible bugs where new test code is only written in one of these. Maybe an env var to switch between the two DB implementations would save us the trouble?

Reply:
Currently we don't support env var for unit testing. If we want to do that we will have to:
1. Change current behavior to use the env var instead of running everything in 2 falvors
2. Add another job in the pipeline to run the unit tests with this env var activated.
I think its too much effort for a temporary need.
On this specific module there were far too many changes to make to support KV enabled mode, so I thought it was better to just duplicate the test module. If you think it's worth the work I can change this module to use the same conventions we used so far for unit testing with KV